src/pubsub/streaming_subscription.rs

```diff
@@ -303,18 +303,20 @@ impl<C> StreamSubscription<C> {
         subscription: String,
         config: StreamSubscriptionConfig,
     ) -> Self {
-        StreamSubscription {
+        let sub = StreamSubscription {
             state: StreamState::Initialized {
                 client,
-                subscription,
+                subscription: subscription.clone(),
                 config,
                 retry_policy: ExponentialBackoff::new(
                     PubSubRetryCheck::default(),
                     Self::default_retry_configuration(),
                 ),
             },
             _p: std::marker::PhantomPinned,
-        }
+        };
+        debug!("Created subscription for topic: {}", subscription);
```

Comment:
if you move this event above the initializer you could borrow the subscription instead of cloning it